metaseq/cli/train.py

```diff
@@ -495,6 +495,57 @@ def post_checkpoint_callback(cfg, filename):
                 ),
             )
             os.remove(filename)
+
+            # Run evals after all checkpoint parts have been uploaded
+            checkpoint_id = checkpoint_dir.split("_")[-1]
+            if (
+                cfg.checkpoint.cloud_eval_script_path is not None
+                and distributed_utils.get_global_rank() == 0
```

Comment:
Noob question: When running on many nodes, does this only run once?

Reply:
Yes